tpm2/tpm2_test.go

```diff
@@ -261,3 +264,128 @@ func TestHash(t *testing.T) {
 		t.Errorf("Hash(%q) returned %x, want %x", val, got, want)
 	}
 }
+
+func TestLoadExternalPublicKey(t *testing.T) {
+	pk, err := rsa.GenerateKey(rand.Reader, 2048)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	rw := openTPM(t)
+	defer rw.Close()
+
+	rp := RSAParams{
+		EncAlg:     AlgRSA,
+		HashAlg:    AlgSHA1,
+		Attributes: FlagSign | FlagSensitiveDataOrigin | FlagUserWithAuth,
+		SymAlg:     AlgNull,
+		Scheme:     AlgRSASSA,
+		SchemeHash: AlgSHA1,
+		ModSize:    2048,
+		Exp:        uint32(pk.PublicKey.E),
+		Modulus:    pk.PublicKey.N.Bytes(),
+	}
+	private := Private{
+		Type:      AlgRSA,
+		Sensitive: pk.Primes[0].Bytes(),
+	}
+	h, _, err := LoadExternal(rw, rp, private, HandleNull)
+	if err != nil {
+		t.Fatal(err)
+	}
+	defer FlushContext(rw, h)
+}
+
+func TestCertify(t *testing.T) {
+	rw := openTPM(t)
+	defer rw.Close()
+
+	params := RSAParams{
+		EncAlg:     AlgRSA,
+		HashAlg:    AlgSHA256,
+		Attributes: FlagSignerDefault,
+		SymAlg:     AlgNull,
+		Scheme:     AlgRSASSA,
+		SchemeHash: AlgSHA256,
+		ModSize:    1024,
+	}
+	signerHandle, signerPub, err := CreatePrimary(rw, HandleOwner, pcrSelection, "", defaultPassword, params)
```

Comment:
[optional]: Just thought of this now: maybe the empty strings could be given a variable in a var section at the top of the test file so that the name of this variable makes it clear what the parameter does and why it would be the empty string.

Reply:
Done